IPNI_HTTP_PROVIDER.md

```diff
@@ -0,0 +1,284 @@
+
+* `application/json` is considered to be the same as `application/vnd.ipld.dag-json`.
+* `application/cbor` is considered to be the same as `application/vnd.ipld.dag-cbor`.
+
+The common data types, such as `CID` and `Multihash`, are encoded in their
+standard [multibase format](https://github.com/multiformats/multibase) when referenced in URLs.
+
+### Versioning
+
+The IPNI Provider HTTP API adopts a versioning strategy that utilizes the path prefix to indicate major releases. As a
+best practice, future iterations that introduce breaking changes should increment the version number in the API path.
+This approach ensures compatibility and allows clients to adapt to new versions while maintaining backward compatibility
+with existing implementations. By explicitly incorporating versioning in the API path, it becomes easier to manage and
+track changes over time, facilitating smooth transitions and providing a clear indication of API evolution.
+
+## Specification
+
+### GET `/ipni/v1/ad/head`
+
+This endpoint retrieves the latest advertisement CID published by the provider, along with additional information such
+as the provider's public key and the topic on which the advertisement is published.
+
+#### Response
+
+The response from the `/ipni/v1/ad/head` endpoint includes the following fields:
+
+- **`head`** (required): The CID of the latest advertisement published by the provider.
+- **`topic`** (optional): The topic name on which the advertisement is published. If not specified, the default value
+  of `/indexer/ingest/mainnet` is assumed.
+- **`pubkey`** (required): The serialized public key of the provider in protobuf format using the libp2p standard. The
+  public key can be marshalled using
+  the [`crypto.MarshalPublicKey`](https://pkg.go.dev/github.com/libp2p/go-libp2p-core/crypto#MarshalPublicKey) function.
+- **`sig`** (required): The signature associated with the `head` CID, obtained by concatenating the bytes of the `head`
+  CID and the UTF-8 bytes of the `topic` (if present). The signature is verified against the `pubkey`.
+
+Please note that the response provides the necessary information to validate the authenticity of the advertisement CID
+and verify its integrity using the provider's public key and the associated signature.
+
+The following snippet represents the IPLD schema of the signed head advertisement:
+
+```ipldsch
+type SignedHead  struct {
+    head   Link
+    topic  optional String
+    pubkey Bytes
+    sig    Bytes
+}
+```
+
+#### Example
+
+The following represents an abbreviated example of head advertisement response in `DAG-JSON` encoding:
+
+```json
+{
+  "head": {
+    "/": "baguqeerazx6qhbdzckrnuwvl3reqnrpysz3ry5qtdxzpo23losoeiywij65a"
+  },
+  "topic": "/indexer/ingest/devnet",
+  "pubkey": {
+    "/": {
+      "bytes": "CAASpgIwggEiMA0G..."
+    }
+  },
+  "sig": {
+    "/": {
+      "bytes": "LrZiicKdqDqkG2UFR..."
+    }
+  }
+}
+```
+
+### GET `/ipni/v1/ad/{CID}`
```

Comment:
Would it be helpful to have bulk fetches in the specification? Something like `GET /ipni/v1/ad/{CID}?depth=20`, so that chains can be traversed faster. Support for depth can be optional too.

Reply:
Out of scope for this spec IMO. 

How it occurs for me is that future iterations can introduce this sort of functionality, along with support for `[application/vnd.ipld.car](https://www.iana.org/assignments/media-types/application/vnd.ipld.car)` where we can ask a provider to preassemble CAR files containing ads.

This functionality is sufficiently complex that deserves its own spec/iteration IMHO. 

Cc @gammazero

Reply:
Sounds good, this is more of a optimisation rather than a must have. Other than that - looks good to me.

Reply:
Added this as an item in [future considerations](https://github.com/ipni/specs/blob/masih/http_prov_spec/IPNI_HTTP_PROVIDER.md#future-considerations)